### DISCUSSION TITLE
Cloudcustodian as a wrapper around ansible

### DISCUSSION THREAD

**Discussion by robertdeheer** [cloud-custodian/cloud-custodian#7442] [General]
### Ask your question

I was looking at alternatives to cloudcustodian and realized that ansible has most of the underlying features required to implement a policy engine except the simple except the simple declarative configuration (even simpler than ansible). In theory, it would be possible to translate the cloudcustodian schema to ansible code/api calls. It sure would be powerful.

I may have completely missed the boat on some requirements, so forgive my ignorance.

**Comment by kapilt** [comment]
custodian tries to learn some lessons from ansible to avoid programming logic in yaml (try/except/loops/conditionals/etc). ansible also generally ignores attempts at idempotency wrt to cloud api usage.

perhaps it would be useful to understand what your goal is? could you explain further?

**Comment by robertdeheer** [comment]
I tried to use cc to find and delete gcp storage buckets and found the api
didn't support deleting buckets. It would have been nice to be able to draw
on the existing functionality in ansible.

Rob
